### DISCUSSION TITLE
tightly coupled configuration

### DISCUSSION THREAD

**Discussion by delanym** [jreleaser/jreleaser#444] [Q&A]
Hi. I wanted to try out the jrelease-maven-plugin, starting with making an announcement on Teams for a basic Maven project. I don't need/want it to interface with the repository, package anything, create a changelog etc. But to do this simple task I've had to really change a lot of things like setting properties in the pom (license, developers, url, etc that I'll never need and don't want!) Now I'm stuck with this error

Execution announce of goal org.jreleaser:jreleaser-maven-plugin:0.7.0:announce failed: Cannot parse version '8.0_302'

You've made this project very dynamic, but internally it seems too tightly coupled. Why are you forcing me to have properties in the pom when Maven doesn't? Im not working on an open source project and I'll never set these fields otherwise:

``` xml
  <description>I</description>
  <url>hate</url>
  <licenses>
    <license>
      <name>compulsory</name>
    </license>
  </licenses>
  <developers>
    <developer>
      <name>fields</name>
    </developer>
  </developers>
```

This is my plugin config so far

``` xml
      <plugin>
        <groupId>org.jreleaser</groupId>
        <artifactId>jreleaser-maven-plugin</artifactId>
        <configuration>
          <jreleaser>
            <announce>
              <teams>
                <active>ALWAYS</active>
                <connectTimeout>20</connectTimeout>
                <readTimeout>60</readTimeout>
                <webhook>${env.TEAMS_DISTRIBUTION}</webhook>
                <messageTemplate>teams.tpl</messageTemplate>
              </teams>
            </announce>
            <release>
              <generic>
                <enabled>false</enabled>
                <token>__DO_NOT_SET_HERE__</token>
                <tagName>v1.0.0</tagName>
              </generic>
            </release>
          </jreleaser>
        </configuration>
        <executions>
          <execution>
            <id>changelog</id>
            <goals>
              <goal>changelog</goal>
            </goals>
            <phase>validate</phase>
          </execution>
          <execution>
            <id>announce</id>
            <goals>
              <goal>announce</goal>
            </goals>
            <phase>initialize</phase>
          </execution>
        </executions>
      </plugin>
```

**Comment by aalmiray** [comment]
The reason for required fields is that the project has no clue if you'd like to use `{{projectLicense}}`, `{{projectDescription}}`, or authors, or other basic metadata as part of the variables that may be used to evaluate templates (such as the `teams.tpl` template you have defined as parameter), regardless of the nature of the project (Open Source or commercial).

However it's true that the [current] focus of the tool is closer to Open Source where most of the metadata you mention must be clearly defined. disabling these checks will likely produce invalid Homebrew formulas, Scoop manifests, Docker images, and other files generated by a release.

This being said, perhaps an additional flag may be used to instruct JReleaser to skip most (if not all validations). That way if you're certain you don't need specific metadata you'd be able to tell JReleaser to ignore it. WDYT?

**Reply by delanym** [reply]
I think make it as strict as possible, but no stricter. I've made no indication that Im releasing to homebrew, scoop or docker, so don't bind me to their requirements.

**Reply by aalmiray** [reply]
Correct, however even if no platform packagers were to be used (as it's your case) how would it know that you won't need or use `{{projectLicense}} or authors in the teams.tpl, or any other announcer template for that matter? This is why JReleaser currently errs on the side of caution and makes strict validation.

Adding a flag to bypass strict validation may be the way to go, wouldn't you agree?

**Comment by aalmiray** [comment]
Oh, and more thing, the version number you're using (`8.0_302`) does not match Semantic Version, which is the default setting hence why you're getting an error. Please refer to https://jreleaser.org/guide/latest/configuration/project.html#_version 

You may need to set `<versionPattern>CUSTOM</versionPattern>`.

**Reply by delanym** [reply]
That's the version of the JDK, not the project. The project.version is "4.5.5.0", but Im not even using it. My template file is a single line:
`{{#f_now}}YYYY-MM-dd{{/f_now}}: {{#projectName}}`

**Reply by aalmiray** [reply]
Oh, could you file a ticket with the full stack trace? JReleaser may be parsing the number incorrectly. The stack trace should be found at `target/jreleaser/trace.log`.

**Reply by delanym** [reply]
The trace.log file breaks off abruptly after 34287 bytes (there are many properties in my build). I see also some of the property values are masked ************

**Reply by aalmiray** [reply]
I think you've been affected by https://github.com/jreleaser/jreleaser/issues/387, specifically by the problem fixed by https://github.com/jreleaser/jreleaser/blob/1713602bcceaa3a5fcaaf98fb850395c86c81789/core/jreleaser-utils/src/main/java/org/jreleaser/util/Version.java#L127-L133

This fix is found in the upcoming `v0.8.0` to be released this weekend. We don't post snapshot releases for plugins thus I'm afraid I'll have to ask you to wait a couple of days more, or build your own plugin snapshot to try it out locally, or switch to the CLI and consumer the latest snapshots we provide as part of the `early-access` builds.

**Reply by delanym** [reply]
I tried to build jrelease and I got "Unknown DIRC version 4"

**Reply by aalmiray** [reply]
That's an error message I do not recognize. What OS and JDK (java -version) are you using to build it?

**Reply by aalmiray** [reply]
FWIW the error persists. Filed https://github.com/jreleaser/jreleaser/issues/445

**Comment by delanym** [comment]
JGit doesn't support git index version 4, and my global git config is such
that a version 4 index was created when I cloned the repo. The question is
why do I need jgit to build this project?

**Reply by aalmiray** [reply]
JGit is required by the nemerosa versioning plugin, used to enrich JAR manifests with build specific metadata.
